api/src/pcapi/core/users/api.py

```diff
@@ -1042,8 +1042,10 @@ def search_public_account(search_query: str, order_by: list[str] | None = None)
 
 
 def search_pro_account(search_query: str, order_by: list[str] | None = None) -> BaseQuery:
-    # Any account which is associated with at least one offerer
-    pro_accounts = models.User.query.join(offerers_models.UserOfferer).distinct(models.User.id)
+    # Any account which is associated with at least one offerer or has NON_ATTACHED_PRO role
+    pro_accounts = models.User.query.outerjoin(offerers_models.UserOfferer).filter(
+        sa.or_(sa.not_(offerers_models.UserOfferer.userId.is_(None)), models.User.has_non_attached_pro_role.is_(True))  # type: ignore [attr-defined]
+    )
```

Comment:
(question) Est-ce qu'on ne pourrait pas simplement renvoyer les utilisateurs qui ont le rôle `PRO` ou `NON_ATTACHED_PRO` ? Il me semble que c'est la même chose fonctionnellement et que ça rendrait la requête plus simple et explicite : 

```suggestion
    # Any account which is associated with at least one offerer or has NON_ATTACHED_PRO role
    pro_accounts = models.User.query.filter(
        models.User.has_non_attached_pro_role.is_(True)
        | models.User.has_pro_role.is_(True)
    )
```


Si on décide de conserver la jointure sur `UserOfferer`, on pourra écrire plus simplement : 

```diff
- sa.not_(offerers_models.UserOfferer.userId.is_(None))
+ offerers_models.UserOfferer.userId.isnot(None)
```

... et même utiliser `|` au lieu de `sa.or_()`.